spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRevertDeleteProcedure.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.extensions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import org.apache.iceberg.DataOperations;
+import org.apache.iceberg.ParameterizedTestExtension;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.expressions.Expressions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ParameterizedTestExtension.class)
+public class TestRevertDeleteProcedure extends ExtensionsTestBase {
+
+  @AfterEach
+  public void removeTable() {
+    sql("DROP TABLE IF EXISTS %s", tableName);
+  }
+
+  private Table initTable() {
+    sql("CREATE TABLE %s (id bigint NOT NULL, data string) USING iceberg", tableName);
+    sql("INSERT INTO %s VALUES (1, 'a')", tableName);
+    sql("INSERT INTO %s VALUES (2, 'b')", tableName);
+    sql("DELETE FROM %s WHERE id = 1", tableName);
+
+    Table table = validationCatalog.loadTable(tableIdent);
+
+    assertThat(spark.table(tableName).count()).isEqualTo(1);
+    assertThat(table.snapshots()).hasSize(3);
+    assertThat(table.currentSnapshot().operation()).isEqualTo(DataOperations.DELETE);
+
+    return table;
+  }
+
+  @TestTemplate
+  public void testRevertDelete() {
+    Table table = initTable();
+    long snapshotId = table.currentSnapshot().snapshotId();
+
+    List<Object[]> output =
+        sql("CALL %s.system.revert_delete('%s', %d)", catalogName, tableIdent, snapshotId);
+
+    assertThat(output.get(0)[0]).isEqualTo(1L);
+
+    // check that the data was added back...
+    table.refresh();
+    assertThat(spark.table(tableName).count()).isEqualTo(2);
+    assertThat(table.snapshots()).hasSize(4);
+    assertThat(table.currentSnapshot().operation()).isEqualTo(DataOperations.APPEND);
```

Comment:
For the data files that were re-added, does the sequence number stay the same?  I think we should check that too so that dropping and adding files doesn't conflict with other positional deletes or equality deletes.